api/v1alpha1/common_types.go

```diff
@@ -134,5 +139,8 @@ type ChaosInstance struct {
 // ChaosList defines a common interface for chaos lists
 type ChaosList interface {
 	runtime.Object
+	metav1.ListInterface
+	// TODO: refactor: move this DeepCopyList() to upper level, rename to SpawnList()
```

Comment:
what is the `upper level`? And why need to rename it?

Reply:
not only rename it, but this method should also be moved to the "upper level", because developers usually forget to use deepcopy, and the origin struct in the global map will be dirty. 

for example, the common usage about this `ChaosList` is mostly like:


https://github.com/chaos-mesh/chaos-mesh/blob/984b3b3140e9f9b11959c8ae922a66d5f6ab26cf/controllers/schedule/utils/list.go#L33-L39

we fetch the `ChaosKind` from the global map then get the `ChaosList`, then invoke the `DeepCopyList()` manually. If some developer forgets to invoke `DeepCopyList()`, bugs MIGHT BE created, but it's hard to find because no compiler error will be throw.

I want remove this `DeepCopyList()` for `ChaosList`, and add new method called `SpawnNewList()`( or `SpawnDeepCopiedList()`) for the struct `ChaosKind`, so developers do NOT need call `DeepCopyList()` manually.

Same manually `DeepCopy()` situation also happens on the `Chaos client.Object` in `ChaosKind`:

```golang
type ChaosKind struct {
	Chaos client.Object // <- I mean this
	ChaosList
}
```

Reply:
tldr; I want:

```golang
type ChaosKind struct {
	chaos client.Object // private
	list ChaosList // private
}

func (it *ChaosKind) SpawnObject() client.Object {
	// implementation
}

func (it *ChaosKind) SpawnList() ChaosList {
	// implementation
}
```

@FingerLeader , if you have interest in this thing, you could have a try and raise another PR after this PR merged. :heart:

Reply:
OK, I think I can do this.